packages/neon_framework/lib/src/utils/request_manager.dart

```diff
@@ -125,36 +128,43 @@ class RequestManager {
         emitEmptyCache,
       );
 
-  Future<void> _wrap<T, R>(
+  /// Executes a generic request.
+  ///
+  /// This method is only meant to be used in testing.
+  @visibleForTesting
+  Future<void> wrap<T, R>(
     String clientID,
     String k,
     BehaviorSubject<Result<T>> subject,
     NextcloudApiCallback<R> call,
     UnwrapCallback<T, R> unwrap,
     SerializeCallback<R> serialize,
     DeserializeCallback<R> deserialize, [
+    // ignore: avoid_positional_boolean_parameters
     bool disableTimeout = false,
     bool emitEmptyCache = false,
+    Duration timeLimit = kDefaultTimeout,
```

Comment:
Why not call it timeout? It's much more common and this name is a bit confusing.

Reply:
because the method is already called like that so the parameter would shadow the method.

Reply:
Ah, then the method should be renamed to something better

Reply:
Dart calls the method timeout and the parameter timeLimit (https://api.dart.dev/stable/3.2.5/dart-async/Future/timeout.html).
I'd rather stick with this.

Reply:
Ok sure :)